### PR TITLE
Remove unnecessary training data from anomaly detection tutorial

### LINKED ISSUE #12705: Anomaly detection pipeline should not use same data for train and test
The spike detection transform does not need training data, so we can provide an empty dataview for that.

The SSA transform does need data but it should be different from the data used to make predictions.

### PR DESCRIPTION
**Pull request by natke**
Preview link: https://review.docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/sales-anomaly-detection?branch=pr-en-us-13675

Fixes #12705

[AB#1572600](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1572600)

[AB#1566041](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1566041)

### CODE REVIEW

**Review by JRAlexander — APPROVED** [review]
LGTM.

File: docs/machine-learning/tutorials/sales-anomaly-detection.md [hunk 1]
```diff
@@ -74,48 +74,48 @@ Add a new class to your project:
...
+   ```
 
-Add the following additional `using` statements to the top of the *Program.cs* file:
+4. Remove the existing class definition and add the following code, which has two classes `ProductSalesData` and `ProductSalesPrediction`, to the *ProductSalesData.cs* file:
 
-[!code-csharp[AddUsings](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#AddUsings "Add necessary usings")]
+    [!code-csharp[DeclareTypes](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/ProductSalesData.cs#DeclareTypes "Declare data record types")]
 
-You need to create two global fields to hold the recently downloaded dataset file path and the saved model file path:
+    `ProductSalesData` specifies an input data class. The [LoadColumn](xref:Microsoft.ML.Data.LoadColumnAttribute.%23ctor%28System.Int32%29) attribute specifies which columns (by column index) in the dataset should be loaded.
```
Comment (luisquintanilla) [inline docs/machine-learning/tutorials/sales-anomaly-detection.md]: Suggestion:

Since input data class is explained, also explain ProductSalesPrediction.

**Review by luisquintanilla — APPROVED** [review]
Looks good. Minor suggestion.

File: docs/machine-learning/tutorials/sales-anomaly-detection.md (hunk 1 above)
Comment (natke) [reply docs/machine-learning/tutorials/sales-anomaly-detection.md]: Good catch. Will do

File: docs/machine-learning/tutorials/sales-anomaly-detection.md [hunk 2]
```diff
@@ -66,144 +66,152 @@ The following table is a data preview from your \*.csv file:
...
-Create a display header line using the following <xref:System.Console.WriteLine?displayProperty=nameWithType> code:
+    [!code-csharp[DisplayHeader1](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#DisplayHeader1)]
 
-[!code-csharp[DisplayHeader1](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#DisplayHeader1)]
+    You'll display the following information in your spike detection results:
 
-You'll display the following information in your spike detection results:
+    * `Alert` indicates a spike alert for a given data point.
+    * `Score` is the `ProductSales` value for a given data point in the dataset.
+    * `P-Value` The "P" stands for probability. The closer to 0, the p-value is the more likely the data point is an anomaly.
```
Comment (luisquintanilla) [inline docs/machine-learning/tutorials/sales-anomaly-detection.md]:
```suggestion
    * `P-Value` The "P" stands for probability. The closer the p-value is to 0, the more likely the data point is an anomaly.
```

File: docs/machine-learning/tutorials/sales-anomaly-detection.md [hunk 3]
```diff
@@ -256,59 +264,59 @@ Alert   Score   P-Value
...
-[!code-csharp[CreateEnumerable2](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#CreateEnumerable2)]
+    [!code-csharp[DisplayHeader2](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#DisplayHeader2)]
 
-Create a display header with the following code as the next line in the `DetectChangePoint()` method:
+    You'll display the following information in your change point detection results:
 
-[!code-csharp[DisplayHeader2](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#DisplayHeader2)]
+    * `Alert` indicates a change point alert for a given data point.
+    * `Score` is the `ProductSales` value for a given data point in the dataset.
+    * `P-Value` The "P" stands for probability. The close the P-value is to 0, the more likely the data point is an anomaly.
```
Comment (luisquintanilla) [inline docs/machine-learning/tutorials/sales-anomaly-detection.md]:
```suggestion
    * `P-Value` The "P" stands for probability. The closer the P-value is to 0, the more likely the data point is an anomaly.
```

**Review by luisquintanilla — APPROVED** [review]
Good. Just fixed a few things.

**Pull request merged**